server/src/main/java/com/objectcomputing/checkins/security/LocalLoginController.java

```diff
@@ -67,10 +67,10 @@ public LocalLoginController(Authenticator authenticator,
         return authenticationResponseFlowable.map(authenticationResponse -> {
             if (authenticationResponse.isAuthenticated() && authenticationResponse.getUserDetails().isPresent()) {
                 UserDetails userDetails = authenticationResponse.getUserDetails().get();
-                MemberProfile memberProfile = currentUserServices.findOrSaveUser(email, email);
-                String name = memberProfile.getName() != null ? memberProfile.getName() : "";
-                userDetails.setAttributes(Map.of("email", memberProfile.getWorkEmail(), "name", name,
-                        "picture", ""));
+                // The values for attributes email, firstName, lastName match the arguments passed to the call
+                // to method CurrentUserServices.findOrSaveUser in LocalUserPasswordAuthProvider.authenticate
+                userDetails.setAttributes(Map.of("email", email, "firstName", email,
+                        "lastName", email, "picture", ""));
```

Comment:
> So, you are skipping the call to find or save here because it's already happening in the authprovider?
> 
Yes, the call to `findOrSaveUser` here is redundant to me. When user logs in the first time, `findOrSaveUser(email, email)` is called in `LocaUserPasswordAuthProvider.authenticate` which should create a new MemberProfile for this user. The call to `findOrSaveUser` here is used only to get the created member's name, but we already know its name is `email` since `email` was passed as the member name in the first `findOrSaveUser` call. 

> I think I would keep the name logic in place for firstname/lastname and send those over as empty string if they don't exist, rather than sending email. Unless there's something I'm missing here?

As discussed above, `email` is used as the member name in the local login controller. (In the local login web interface, it has 2 fields for email and role. I think those two are used as the username and password in the local environment.) So the name passed to `userDetails.setAttributes()` is always `email` although it isn't obvious in the old code.

Since now we split name to first name/last name but the front-end only sends email and role, I am using the given email as both first name and last name (https://github.com/oci-labs/check-ins/blob/c044a7a0cb216a15b99e64a1d462afe92f999413/server/src/main/java/com/objectcomputing/checkins/security/LocalUserPasswordAuthProvider.java#L39).

Reply:
That's only true if the user isn't in the loaded data. I believe that if the user is in the loaded test data, then this would cause that user not to be recognized as the user entering the system.

Reply:
You're right. I forgot about the loaded data. I've changed this to set the correct first and last names.